### PR TITLE
Cleanup master

### PR DESCRIPTION
**Pull request by alerque**
Fix basic issue with length parser so that tests pass on master again and touch up a few coding paradigms while mucking around.

### CODE REVIEW

**Comment by simoncozens** [comment]
All looks good. You don't have to PR, by the way; it's fine for you to contribute directly to master...

File: core/length.lua [hunk 1]
```diff
@@ -105,6 +105,7 @@ local length = {
   end,
   parse = function (spec)
     if not spec then return _length {} end
+    if type(spec) == "table" then return _length {spec} end
```
Comment (simoncozens) [inline core/length.lua]: Excellent! I needed this anyway to make `\vbox[length=...]` work.

**Comment by alerque** [comment]
Alright thanks I might do that for some minor changes. It would be easier to keep PRs with substantial stuff that needs review free of unrelated clutter if I didn't have to keep moving branch points so often.

**Comment by alerque** [comment]
(Although you  probably should have waited for Travis on this one. I was experimenting with bash syntax and seem to have broken it.)

**Comment by simoncozens** [comment]
I've fixed it, I think.

**Comment by alerque** [comment]
Nope. You fixed the other thing.

**Pull request merged**